### ISSUE TITLE
Question regarding getGrade() for Lab 6 level 3

### ISSUE THREAD

**Issue by kyanroh** [nus-cs2030/2122-s2#135]
# Summary

I'm trying to complete level 6 and making my roster but im not too sure how to go about completing the section on getGrade().

## Code Snippets

String getGrade(String studentName, String moduleName, String assessmentName) {
       return this.get(studentName).get(moduleName).get(assessmentName).getGrade();
However, this does not work as it returns me an optional.

How should i go about getting the grade?

**Comment by AmirulShyakir** [comment]
Hi, you can take a look at the test cases for level 3 right before getGrade() it shows how to chain the flatMaps to obtain a grade wrapped in an optional. Using this, you can chain with an orElse(string) to obtain the grade should the conditions of the query exist, otherwise it returns the string for no such record.

**Comment by jlee1007** [comment]
Yes you can reference from the test cases in level 3 & create an Optional<String>. The main thing you would need to edit is the Assessment::getGrade() as CC will fail that due to method reference.

**Comment by rexlionz** [comment]
Hi! I believe that you are on the right track, but with your code you are bound to face the issue if the student name/module name/ assessment name is incorrect, then based on how Immutable Map is implemented you would get an optional empty. As what Aimirul mentioned, you should use orElse to return you the string implementation that the level wants should the record not exist.

**Comment by HaiJin123** [comment]
I think you should use map functions to fix this problem.

**Comment by undeser** [comment]
Hi Ryan, try making use of the ```flatMap``` and ```map``` methods. I believe you got the rough idea there already.
